plugins/woocommerce-admin/client/marketplace/contexts/marketplace-context.tsx

```diff
@@ -59,6 +71,8 @@ export function MarketplaceContextProvider( props: {
 		addInstalledProduct,
 		hasBusinessServices,
 		setHasBusinessServices,
+		searchResultsCount,
+		setSearchResultsCount,
```

Comment:
idea/proposal, do as you wish:

I wonder if this code couldn't be reworked a bit, to make it easier to use `setSearchResultsCount`. I see that previous value is not really needed when calling it, so how about doing a couple of tweaks here in this file similar to:

1. Renaming state to `const [ searchResultsCounts, setSearchResultsCountState ] = useState(...)`
2. Adding a callback function
```
const setSearchResultsCount = useCallback(( updatedCounts) =>
   setSearchResultsCountState( prev => ({ ...prev, updatedCounts } );
, []);
```

Then, whenever `setSearchResultsCount` is called it can be passed just an object as a parameter, not the whole callback function.

Reply:
> I see that previous value is not really needed when calling it

`prevState` is needed (in the places where we're using it), but I think I get your idea about this wrapper. It makes it much easier to use if we don't expose it.

Refactored in 9ebdb80359